src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,31 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int USER_MINIMUM_AGE = 18;
+    public static final int PASSWORD_MINIMUM_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null || user.getLogin() == null || user.getPassword() == null
+                || user.getAge() == null) {
+            throw new RegistrationServiceException("User fields can't be null/empty/negative");
+        }
+        if (user.getAge() < 0 || user.getLogin().isEmpty()) {
```

Comment:
```suggestion
        if (user.getAge() < 0) {
```
this "user.getLogin().isEmpty()" you checked in first if